From 809b4acadba552fe1c3afaa9441e29c2f225d174 Mon Sep 17 00:00:00 2001
From: "Badrappan, Jeevaka" <jeevaka.badrappan@intel.com>
Date: Fri, 16 Jun 2023 22:55:48 +0530
Subject: [PATCH] Remove aidl, libminigbm_gralloc_* and gralloc.* Android.bp
 files

Removed Anrdroid.bp files to make sure aidl, libminigbm_gralloc_* and
gralloc.* are not compiled from AOSP tree.

Tracked-On: OAM-110710
Change-Id: I7ed204e1200b3174720d655e1514245de91f7774
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 Android.bp                             | 262 -------------------------
 cros_gralloc/aidl/Android.bp           |  51 -----
 cros_gralloc/mapper_stablec/Android.bp |  44 -----
 3 files changed, 357 deletions(-)
 delete mode 100644 Android.bp
 delete mode 100644 cros_gralloc/aidl/Android.bp
 delete mode 100644 cros_gralloc/mapper_stablec/Android.bp

diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index e2d74d2..0000000
--- a/Android.bp
+++ /dev/null
@@ -1,262 +0,0 @@
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-package {
-    default_applicable_licenses: ["external_minigbm_license"],
-}
-
-// Added automatically by a large-scale-change that took the approach of
-// 'apply every license found to every target'. While this makes sure we respect
-// every license restriction, it may not be entirely correct.
-//
-// e.g. GPL in an MIT project might only apply to the contrib/ directory.
-//
-// Please consider splitting the single license below into multiple licenses,
-// taking care not to lose any license_kind information, and overriding the
-// default license using the 'licenses: [...]' property on targets as needed.
-//
-// For unused files, consider creating a 'fileGroup' with "//visibility:private"
-// to attach the license to, and including a comment whether the files may be
-// used in the current project.
-// See: http://go/android-license-faq
-license {
-    name: "external_minigbm_license",
-    visibility: [":__subpackages__"],
-    license_kinds: [
-        "SPDX-license-identifier-Apache-2.0",
-        "SPDX-license-identifier-BSD",
-        "SPDX-license-identifier-MIT",
-    ],
-    license_text: [
-        "LICENSE",
-    ],
-}
-
-filegroup {
-    name: "minigbm_core_files",
-
-    srcs: [
-        "amdgpu.c",
-        "drv.c",
-        "drv_array_helpers.c",
-        "drv_helpers.c",
-        "dumb_driver.c",
-        "i915.c",
-        "mediatek.c",
-        "msm.c",
-        "rockchip.c",
-        "vc4.c",
-        "virtgpu.c",
-        "virtgpu_cross_domain.c",
-        "virtgpu_virgl.c",
-    ],
-}
-
-filegroup {
-    name: "minigbm_gralloc_common_files",
-
-    srcs: [
-        "cros_gralloc/cros_gralloc_buffer.cc",
-        "cros_gralloc/cros_gralloc_helpers.cc",
-        "cros_gralloc/cros_gralloc_driver.cc",
-    ],
-}
-
-filegroup {
-    name: "minigbm_gralloc0_files",
-    srcs: ["cros_gralloc/gralloc0/gralloc0.cc"],
-}
-
-cc_defaults {
-    name: "minigbm_defaults",
-
-    cflags: [
-        "-D_GNU_SOURCE=1",
-        "-D_FILE_OFFSET_BITS=64",
-        "-Wall",
-        "-Wsign-compare",
-        "-Wpointer-arith",
-        "-Wcast-qual",
-        "-Wcast-align",
-        "-Wno-unused-parameter",
-    ],
-
-    product_variables: {
-        platform_sdk_version: {
-            cflags: ["-DANDROID_API_LEVEL=%d"],
-        },
-    },
-}
-
-cc_library_headers {
-    name: "minigbm_headers",
-    host_supported: true,
-    vendor_available: true,
-    export_include_dirs: ["."],
-}
-
-cc_defaults {
-    name: "minigbm_cros_gralloc_defaults",
-
-    defaults: ["minigbm_defaults"],
-
-    header_libs: [
-        "libhardware_headers",
-        "libnativebase_headers",
-        "libsystem_headers",
-        "minigbm_headers",
-    ],
-
-    static_libs: ["libarect"],
-
-    vendor: true,
-
-    shared_libs: [
-        "libcutils",
-        "libdmabufheap",
-        "libdrm",
-        "libnativewindow",
-        "libsync",
-        "liblog",
-    ],
-}
-
-cc_defaults {
-    name: "minigbm_cros_gralloc_library_defaults",
-
-    defaults: ["minigbm_cros_gralloc_defaults"],
-    srcs: [
-        ":minigbm_core_files",
-        ":minigbm_gralloc_common_files",
-    ],
-}
-
-cc_defaults {
-    name: "minigbm_cros_gralloc0_defaults",
-
-    defaults: ["minigbm_cros_gralloc_defaults"],
-    relative_install_path: "hw",
-
-    srcs: [":minigbm_gralloc0_files"],
-}
-
-cc_library {
-    name: "libgbm",
-    defaults: ["minigbm_defaults"],
-    host_supported: true,
-
-    srcs: [
-        ":minigbm_core_files",
-        "gbm.c",
-        "gbm_helpers.c",
-    ],
-
-    target: {
-        host: {
-            // Avoid linking to another host copy of libdrm; this library will cause
-            // binary GPU drivers to be loaded from the host, which might be linked
-            // to a system copy of libdrm, which conflicts with the AOSP one
-            allow_undefined_symbols: true,
-            header_libs: ["libdrm_headers"],
-        },
-        android: {
-            shared_libs: [
-                "libdrm",
-                "liblog"
-            ],
-        },
-    },
-    apex_available: [
-        "//apex_available:platform",
-        "com.android.virt",
-    ],
-
-    export_include_dirs: ["."],
-}
-
-// Generic
-cc_library_shared {
-    name: "libminigbm_gralloc",
-    defaults: ["minigbm_cros_gralloc_library_defaults"],
-    cflags: ["-DHAS_DMABUF_SYSTEM_HEAP"],
-}
-
-cc_library_shared {
-    name: "gralloc.minigbm",
-    defaults: ["minigbm_cros_gralloc0_defaults"],
-    shared_libs: ["libminigbm_gralloc"],
-}
-
-// Intel
-cc_library_shared {
-    name: "libminigbm_gralloc_intel",
-    defaults: ["minigbm_cros_gralloc_library_defaults"],
-    cflags: ["-DDRV_I915"],
-    enabled: false,
-    arch: {
-        x86: {
-            enabled: true,
-        },
-        x86_64: {
-            enabled: true,
-        },
-    },
-}
-
-cc_library_shared {
-    name: "gralloc.minigbm_intel",
-    defaults: ["minigbm_cros_gralloc0_defaults"],
-    shared_libs: ["libminigbm_gralloc_intel"],
-    enabled: false,
-    arch: {
-        x86: {
-            enabled: true,
-        },
-        x86_64: {
-            enabled: true,
-        },
-    },
-}
-
-// Meson
-cc_library_shared {
-    name: "libminigbm_gralloc_meson",
-    defaults: ["minigbm_cros_gralloc_library_defaults"],
-    cflags: ["-DDRV_MESON"],
-}
-
-cc_library_shared {
-    name: "gralloc.minigbm_meson",
-    defaults: ["minigbm_cros_gralloc0_defaults"],
-    shared_libs: ["libminigbm_gralloc_meson"],
-}
-
-// MSM
-cc_library_shared {
-    name: "libminigbm_gralloc_msm",
-    defaults: ["minigbm_cros_gralloc_library_defaults"],
-    cflags: [
-        "-DDRV_MSM",
-        "-DQCOM_DISABLE_COMPRESSED_NV12",
-        "-DHAS_DMABUF_SYSTEM_HEAP",
-    ],
-}
-
-cc_library_shared {
-    name: "gralloc.minigbm_msm",
-    defaults: ["minigbm_cros_gralloc0_defaults"],
-    shared_libs: ["libminigbm_gralloc_msm"],
-}
-
-// ARCVM
-cc_library_shared {
-    name: "libminigbm_gralloc_arcvm",
-    defaults: ["minigbm_cros_gralloc_library_defaults"],
-    cflags: ["-DVIRTIO_GPU_NEXT"],
-}
-
-cc_library_shared {
-    name: "gralloc.minigbm_arcvm",
-    defaults: ["minigbm_cros_gralloc0_defaults"],
-    shared_libs: ["libminigbm_gralloc_arcvm"],
-}
diff --git a/cros_gralloc/aidl/Android.bp b/cros_gralloc/aidl/Android.bp
deleted file mode 100644
index da8f670..0000000
--- a/cros_gralloc/aidl/Android.bp
+++ /dev/null
@@ -1,51 +0,0 @@
-//
-// Copyright (C) 2022 The Android Open Source Project
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package {
-    // See: http://go/android-license-faq
-    // A large-scale-change added 'default_applicable_licenses' to import
-    // all of the 'license_kinds' from "external_minigbm_license"
-    // to get the below license kinds:
-    //   SPDX-license-identifier-Apache-2.0
-    //   SPDX-license-identifier-BSD
-    default_applicable_licenses: ["external_minigbm_license"],
-}
-
-cc_binary {
-    name: "android.hardware.graphics.allocator-service.minigbm",
-    defaults: ["minigbm_cros_gralloc_defaults"],
-    relative_install_path: "hw",
-    init_rc: ["allocator.rc"],
-    vintf_fragments: ["allocator.xml"],
-    vendor: true,
-    shared_libs: [
-        "android.hardware.graphics.allocator-V2-ndk",
-        "android.hardware.graphics.mapper@4.0",
-        "libbase",
-        "libbinder_ndk",
-        "libgralloctypes",
-        "libhidlbase",
-        "liblog",
-        "libminigbm_gralloc",
-        "libminigbm_gralloc4_utils",
-    ],
-    static_libs: [
-        "libaidlcommonsupport",
-    ],
-    srcs: [
-        "Allocator.cpp",
-        "Main.cpp",
-    ],
-}
diff --git a/cros_gralloc/mapper_stablec/Android.bp b/cros_gralloc/mapper_stablec/Android.bp
deleted file mode 100644
index b25be17..0000000
--- a/cros_gralloc/mapper_stablec/Android.bp
+++ /dev/null
@@ -1,44 +0,0 @@
-//
-// Copyright (C) 2022 The Android Open Source Project
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package {
-    // See: http://go/android-license-faq
-    // A large-scale-change added 'default_applicable_licenses' to import
-    // all of the 'license_kinds' from "external_minigbm_license"
-    // to get the below license kinds:
-    //   SPDX-license-identifier-Apache-2.0
-    //   SPDX-license-identifier-BSD
-    default_applicable_licenses: ["external_minigbm_license"],
-}
-
-cc_library_shared {
-    name: "mapper.minigbm",
-    defaults: ["minigbm_gralloc4_common_defaults"],
-    vintf_fragments: ["mapper.minigbm.xml"],
-    shared_libs: [
-        "android.hardware.graphics.allocator-V2-ndk",
-        "libminigbm_gralloc",
-    ],
-    header_libs: [
-        "libbase_headers",
-        "libimapper_stablec",
-        "libimapper_providerutils",
-    ],
-    srcs: [
-        ":minigbm_gralloc4_mapper_files",
-        "Mapper.cpp",
-    ],
-    cpp_std: "c++20",
-}
-- 
2.40.0

